librz/include/rz_type.h

```diff
@@ -41,33 +41,35 @@ typedef struct rz_type_db_t {
 
 // Base types
 
+typedef enum {
+	RZ_BASE_TYPE_KIND_STRUCT,
+	RZ_BASE_TYPE_KIND_UNION,
+	RZ_BASE_TYPE_KIND_ENUM,
+	RZ_BASE_TYPE_KIND_TYPEDEF, // probably temporary addition, dev purposes
+	RZ_BASE_TYPE_KIND_ATOMIC, // For real atomic base types
+} RzBaseTypeKind;
+
 typedef struct rz_type_enum_case_t {
 	char *name;
 	int val;
 } RzTypeEnumCase;
 
 typedef struct rz_type_struct_member_t {
+	RzBaseTypeKind kind; // Any member could be of any possible type kind
```

Comment:
I don't think we should or even can do it like this. Due to how the sdb works, we should probably only allow references to other nested structs/types by name so if you have an anonymous nested struct, you have to assign it a name and refer to it by that.